lib/foreman/http_proxy/net_http_extension.rb

```diff
@@ -2,9 +2,8 @@ module Foreman
   module HttpProxy
     module NetHttpExtension
       def request(req, body = nil, &block)
-        host = URI.parse(@address).host
-        if proxy_http_request?(@proxy_address, host, @socket)
-          log_proxied_request("NetHttp", http_proxy, host)
+        if proxy_http_request?(@proxy_address, @address, @socket)
```

Comment:
How did this ever work? The signature is here:
https://github.com/theforeman/foreman/blob/44764290e760111db7c41b859cc242a74e3516ed/lib/foreman/http_proxy.rb#L16

So the third argument is a schema, but a socket is passed in? How is that ever true?

Reply:
Hah lemme see, it's Monday so I should be fresh now.

Reply:
Correction, it's Tuesday. Yay! :-)

Anyway, it looks like @socket is either:

1. `nil` if you are doing non-persistent connection, which by default are all connections unless you call Net::HTTP.start with a block
2. `false` in all other cases:

    ['http', 'https'].include?(Net::BufferedIO.new(StringIO.new))

Conclusion: this has never worked.

Reply:
To debug this:

```
irb

require "pry"
module NetHttpExtension
   def request(req, body = nil, &block)
      binding.pry
      super(req, body, &block)
    end
 end
Net::HTTP.class_eval do
  prepend NetHttpExtension
end

[9] pry(#<Net::HTTP>)> @socket
=> #<Net::BufferedIO io=#<TCPSocket:0x000000000112ec28>>
[10] pry(#<Net::HTTP>)> ['http', 'https'].include?(@socket)
=> false
```

The correct solution is: `req.uri.scheme.start_with?("http")`

Reply:
Just wondering: when is it ever not a HTTP(S) scheme? When it's some unix socket? I found https://github.com/puppetlabs/net_http_unix so I guess technically that could be it.

Reply:
I have no idea to be honest, this was added from the day one across all three implementations (Net::HTTP, RestClient, Excon) in 7f8fd14d7a7fb4030bc2e6c36a297b4aa864b7ba maybe @bastilian would put some light on it?

Sebastian a question for you: why the code passed URI scheme to the `proxy_http_request?` method? Aren't all three libraries purely for HTTP and therefore this check is not needed?